inc/os_linux.h

```diff
@@ -7,15 +7,36 @@
 //
 // Linux OS api used in PAL implementation
 //
+
+// Include default apis ...
+#if !defined(_DEFAULT_SOURCE)
 #define _DEFAULT_SOURCE 1
 #define _BSD_SOURCE 1
 #define _SVID_SOURCE 1
+#endif
+
+// ... plus xopen and posix apis 
+#if !defined(_XOPEN_SOURCE) || (_XOPEN_SOURCE < 600)
+#undef _XOPEN_SOURCE
 #define _XOPEN_SOURCE 600
+#endif
+
+// ... and any linux specific ones
+#if !defined(_GNU_SOURCE)
 #define _GNU_SOURCE 1
+#endif
+
+// Ensure safe string manipulation ...
+#if !defined(_FORTIFY_SOURCE)
 #define _FORTIFY_SOURCE 1
+#endif
+
+// ... and C'99 compliant source
+#if !defined(_ISOC99_SOURCE)
 #define _ISOC99_SOURCE 1
-#include <features.h>
+#endif
 
+#include <features.h>
```

Comment:
Absolutely.  I was debating this as well.  I ended up preferring to explicitely pull it in and have it evaluated here.  Plus I read the man as that it is "not necessary", i.e. for convinience included by everything else, not that it must not be done.